src/app/tests/TestSceneTable.cpp

```diff
@@ -0,0 +1,852 @@
+static const SceneData sceneData2(CharSpan("Scene #2", sizeof("Scene #2")), 2, 5);
+static const SceneData sceneData3(CharSpan("Scene #3", sizeof("Scene #3")), 25);
+static const SceneData sceneData4(CharSpan("Scene num4", sizeof("Scene num4")), 5);
+static const SceneData sceneData5(CharSpan(), 10);
+static const SceneData sceneData6(CharSpan("Scene #6", sizeof("Scene #6")), 3, 15);
+static const SceneData sceneData7(CharSpan("Scene #7", sizeof("Scene #7")), 20, 5);
+static const SceneData sceneData8(CharSpan("NAME TOO LOOONNG", sizeof("Scene num4")), 10);
+static const SceneData sceneData9(CharSpan("Scene #9", sizeof("Scene #9")), 30, 15);
+static const SceneData sceneData10(CharSpan("Scene #10", sizeof("Scene #10")), 10, 1);
+static const SceneData sceneData11(CharSpan("Scene #11", sizeof("Scene #11")), 20, 10);
+static const SceneData sceneData12(CharSpan("Scene #12", sizeof("Scene #12")), 30, 5);
+
+// Scenes
+SceneTableEntry scene1(sceneId1, sceneData1);
+SceneTableEntry scene2(sceneId2, sceneData2);
+SceneTableEntry scene3(sceneId3, sceneData3);
+SceneTableEntry scene4(sceneId4, sceneData4);
+SceneTableEntry scene5(sceneId5, sceneData5);
+SceneTableEntry scene6(sceneId6, sceneData6);
+SceneTableEntry scene7(sceneId7, sceneData7);
+SceneTableEntry scene8(sceneId8, sceneData8);
+SceneTableEntry scene9(sceneId9, sceneData9);
+SceneTableEntry scene10(sceneId1, sceneData10);
+SceneTableEntry scene11(sceneId5, sceneData11);
+SceneTableEntry scene12(sceneId8, sceneData12);
+
+// Clusters EFS data
+static app::Clusters::Scenes::Structs::ExtensionFieldSet::Type OOextensionFieldSet;
+static app::Clusters::Scenes::Structs::ExtensionFieldSet::Type LCextensionFieldSet;
+static app::Clusters::Scenes::Structs::ExtensionFieldSet::Type CCextensionFieldSet;
+
+static app::Clusters::Scenes::Structs::AttributeValuePair::Type OOPairs[1];
+static app::Clusters::Scenes::Structs::AttributeValuePair::Type LCPairs[2];
+static app::Clusters::Scenes::Structs::AttributeValuePair::Type CCPairs[8];
+
+static uint8_t OO_buffer[scenes::kMaxFieldsPerCluster] = { 0 };
+static uint8_t LC_buffer[scenes::kMaxFieldsPerCluster] = { 0 };
+static uint8_t CC_buffer[scenes::kMaxFieldsPerCluster] = { 0 };
+
+/// @brief Simulates a Handler where Endpoint 1 supports onoff and level control and Endpoint 2 supports onoff and color control
+class TestSceneHandler : public scenes::DefaultSceneHandlerImpl
+{
+public:
+    TestSceneHandler() = default;
+    ~TestSceneHandler() override {}
+
+    // Fills in cluster buffer and adjusts its size to lower than the maximum number of cluster per scenes
+    virtual void GetSupportedClusters(EndpointId endpoint, Span<ClusterId> & clusterBuffer) override
+    {
+        ClusterId * buffer = clusterBuffer.data();
+        if (endpoint == TEST_ENDPOINT1)
+        {
+            if (clusterBuffer.size() >= 2)
+            {
+                buffer[0] = ON_OFF_CID;
+                buffer[1] = LV_CTR_CID;
+                clusterBuffer.reduce_size(2);
```

Comment:
Must check that clusterBuffer.size() big enough before doing this

Reply:
if (clusterBuffer.size() >= 2) does that no?